python/tests/test_scip.py

```diff
@@ -42,3 +42,122 @@ def test_set_param(model, name, param_type):
+    ("limits/nodes", 0.1),
+    ("limits/nodes", "toto"),
+    ("limits/nodes", "a"),
+    ("limits/nodes", ""),
+    ("limits/nodes", None),
+    # Real in [0,1e+20]
+    ("limits/time", -5),
+    ("limits/time", -0.1),
+    ("limits/time", 1e21),
+    ("limits/time", "toto"),
+    ("limits/time", "a"),
+    ("limits/time", ""),
+    ("limits/time", None),
+    # Real in [0,1.79769313486232e+308]
+    ("heuristics/undercover/maxcoversizeconss", -5),
+    ("heuristics/undercover/maxcoversizeconss", -0.1),
+    ("heuristics/undercover/maxcoversizeconss", "toto"),
+    ("heuristics/undercover/maxcoversizeconss", "a"),
+    ("heuristics/undercover/maxcoversizeconss", ""),
+    ("heuristics/undercover/maxcoversizeconss", None),
+    # Char in [d,s,u,v]
+    # ("branching/pscost/strategy", 0), NUL character valid for SCIP ?
+    ("branching/pscost/strategy", 6),
+    ("branching/pscost/strategy", -5),
+    ("branching/pscost/strategy", -0.1),
+    ("branching/pscost/strategy", "toto"),
+    ("branching/pscost/strategy", "a"),
+    ("branching/pscost/strategy", ""),
+    ("branching/pscost/strategy", True),
+    # ("branching/pscost/strategy", False), NUL character valid for SCIP ?
+    ("branching/pscost/strategy", None),
+    # String, combination of 'l', 'n', 'i' characters
+    ("heuristics/undercover/fixingalts", -5),
+    ("heuristics/undercover/fixingalts", -0.1),
+    ("heuristics/undercover/fixingalts", 6),
+    # ("heuristics/undercover/fixingalts", "toto"),  valid for SCIP ?
+    # ("heuristics/undercover/fixingalts", "a"),  valid for SCIP ?
+    ("heuristics/undercover/fixingalts", True),
+    ("heuristics/undercover/fixingalts", False),
+    ("heuristics/undercover/fixingalts", None),
+)
+
+
+@pytest.mark.parametrize("name,value", should_fail)
+def test_set_param_fail(model, name, value):
+    with pytest.raises((ecole.scip.Exception, TypeError)):
+        model.set_param(name, value)
+
+
+should_succeed = (
+    # Bool in [TRUE,FALSE]
+    ("branching/preferbinary", True),
+    ("branching/preferbinary", False),
+    ("branching/preferbinary", 1),
+    ("branching/preferbinary", 0),
+    # Long int in [-1,9223372036854775807]
+    ("limits/nodes", 1e18),
+    ("limits/nodes", 1.0),
+    ("limits/nodes", True),
+    ("limits/nodes", False),
+    # Real in [0,1e+20]
+    ("limits/time", 0),
+    ("limits/time", 0.1),
+    ("limits/time", True),
+    ("limits/time", False),
+    # Real in [0,1.79769313486232e+308]
+    ("heuristics/undercover/maxcoversizeconss", 1e200),
+    # Char in [d,s,u,v]
+    ("branching/pscost/strategy", "d"),
+    ("branching/pscost/strategy", "s"),
+    ("branching/pscost/strategy", "u"),
+    ("branching/pscost/strategy", "v"),
+    ("branching/pscost/strategy", 100),  # do we want to forbid this ?
+    ("branching/pscost/strategy", 115),  # do we want to forbid this ?
+    ("branching/pscost/strategy", 117),  # do we want to forbid this ?
+    ("branching/pscost/strategy", 118),  # do we want to forbid this ?
```

Comment:
Yes, definitely.

Reply:
Then this has to be handled on the Python side, which does not have the concept of a `char`. In C++ there is no way to distinguish between `'d'` and 100.

Reply:
Are all the SCIP char param litteraly characters?
We could further specialize the C++ cast from arithmetic to `char` to
- Convert the `int` 0-9 to the `char` 0-9 ([example](https://stackoverflow.com/questions/4629050/convert-an-int-to-ascii-character))
- Exception on other numbers

Sounds like a separate issue. I would postpone for now and leave it as it is.